service/src/main/java/com/commercetools/pspadapter/payone/mapping/klarna/KlarnaRequestFactory.java

```diff
@@ -0,0 +1,163 @@
+package com.commercetools.pspadapter.payone.mapping.klarna;
+
+import com.commercetools.pspadapter.payone.config.PayoneConfig;
+import com.commercetools.pspadapter.payone.domain.ctp.PaymentWithCartLike;
+import com.commercetools.pspadapter.payone.domain.payone.model.common.AuthorizationRequest;
+import com.commercetools.pspadapter.payone.domain.payone.model.common.ClearingType;
+import com.commercetools.pspadapter.payone.domain.payone.model.klarna.BaseKlarnaRequest;
+import com.commercetools.pspadapter.payone.domain.payone.model.klarna.KlarnaAuthorizationRequest;
+import com.commercetools.pspadapter.payone.domain.payone.model.klarna.KlarnaPreauthorizationRequest;
+import com.commercetools.pspadapter.payone.mapping.CountryToLanguageMapper;
+import com.commercetools.pspadapter.payone.mapping.MappingUtil;
+import com.commercetools.pspadapter.payone.mapping.PayoneRequestFactory;
+import com.commercetools.pspadapter.tenant.TenantConfig;
+import com.google.common.base.Preconditions;
+import io.sphere.sdk.carts.CartLike;
+import io.sphere.sdk.models.Address;
+import io.sphere.sdk.payments.Payment;
+import io.sphere.sdk.types.CustomFields;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import static com.commercetools.pspadapter.payone.mapping.CustomFieldKeys.*;
+import static com.commercetools.pspadapter.payone.mapping.MappingUtil.getFirstValueFromAddresses;
+import static java.util.Arrays.asList;
+
+public class KlarnaRequestFactory extends PayoneRequestFactory {
+
+    @Nonnull
+    private final CountryToLanguageMapper countryToLanguageMapper;
+
+    public KlarnaRequestFactory(@Nonnull final TenantConfig tenantConfig,
+                                @Nonnull final CountryToLanguageMapper countryToLanguageMapper) {
+        super(tenantConfig);
+        this.countryToLanguageMapper = countryToLanguageMapper;
+    }
+
+    @Override
+    public KlarnaPreauthorizationRequest createPreauthorizationRequest(final PaymentWithCartLike paymentWithCartLike) {
+        return createRequestInternal(paymentWithCartLike, KlarnaPreauthorizationRequest::new);
+    }
+
+    /**
+     * @see KlarnaAuthorizationRequest
+     */
+    @Override
+    public KlarnaAuthorizationRequest createAuthorizationRequest(final PaymentWithCartLike paymentWithCartLike) {
+        return createRequestInternal(paymentWithCartLike, KlarnaAuthorizationRequest::new);
+    }
+
+    protected <BKR extends BaseKlarnaRequest> BKR createRequestInternal(final PaymentWithCartLike paymentWithCartLike,
+                                                                        final TriFunction<PayoneConfig, String, PaymentWithCartLike, BKR> requestConstructor) {
+        final Payment ctPayment = paymentWithCartLike.getPayment();
+        Preconditions.checkArgument(ctPayment.getCustom() != null, "Missing custom fields on payment!");
```

Comment:
it will be printed by exception catcher: https://github.com/commercetools/commercetools-payone-integration/blob/0a1ca55e4fda35ac165790beb7e610e67cd178e7/service/src/main/java/com/commercetools/pspadapter/payone/PaymentHandler.java#L78